camel/retrievers/hybrid_retrival.py

```diff
@@ -0,0 +1,238 @@
+        rank_smoothing_factor: int = 60,
+        vector_retriever_top_k: int = 50,
+        vector_retriever_similarity_threshold: float = 0.5,
+        bm25_retriever_top_k: int = 50,
+        return_detailed_info: bool = False,
+    ) -> Union[
+        dict[str, Sequence[Collection[str]]], dict[str, Sequence[str | float]]
+    ]:
+        r"""Executes a hybrid retrieval query using both vector and BM25
+        retrievers.
+
+        Args:
+            query (str): The search query.
+            top_k (int): Number of top results to return (default 20).
+            vector_weight (float): Weight for vector retriever results in RRF.
+            bm25_weight (float): Weight for BM25 retriever results in RRF.
+            rank_smoothing_factor (int): RRF hyperparameter for rank smoothing.
+            vector_retriever_top_k (int): Top results from vector retriever.
+            vector_retriever_similarity_threshold (float): Similarity
+                threshold for vector retriever.
+            bm25_retriever_top_k (int): Top results from BM25 retriever.
+            return_detailed_info (bool): Return detailed info if True.
+
+        Returns:
+            dict[str, Sequence[Union[Collection[str], str, float]]]: By
+                default, returns only the text information. If
+                `return_detailed_info` is `True`, return detailed information
+                including rrf scores.
+        """
+        if top_k > max(vector_retriever_top_k, bm25_retriever_top_k):
+            raise ValueError(
+                "top_k needs to be less than or equal to the "
+                "maximum value among vector_retriever_top_k and "
+                "bm25_retriever_top_k."
+            )
+        if vector_weight < 0 or bm25_weight < 0:
+            raise ValueError(
+                "Neither `vector_weight` nor `bm25_weight` can be negative."
+            )
+
+        vector_retriever_results = self.auto_retriever.run_vector_retriever(
+            query=query,
+            contents=self.content_input_path,
+            top_k=vector_retriever_top_k,
+            similarity_threshold=vector_retriever_similarity_threshold,
+            return_detailed_info=True,
+        )
+        bm25_retriever_results = self.bm25_retriever.query(
+            query=query,
+            top_k=bm25_retriever_top_k,
+        )
+
+        all_retrieved_info = self._sort_rrf_scores(
+            vector_retriever_results,
+            bm25_retriever_results,
+            top_k,
+            vector_weight,
+            bm25_weight,
+            rank_smoothing_factor,
+        )
+
+        detailed_info = {
+            "Original Query": query,
+            "Retrieved Context": all_retrieved_info,
+        }
+
+        text_retrieved_info = [item['text'] for item in all_retrieved_info]
+
+        text_info = {
+            "Original Query": query,
+            "Retrieved Context": text_retrieved_info,
+        }
+        if return_detailed_info:
```

Comment:
The logic in this section can be simplified to avoid redundancy.
```suggestion
       retrieved_info = {
    "Original Query": query,
    "Retrieved Context": all_retrieved_info if return_detailed_info else [item['text'] for item in all_retrieved_info],
}
return retrieved_info
```